threat-model/readme.md

```diff
@@ -99,85 +93,159 @@ See [section 2. First Parties, Embedded Parties and Delegated Parties](#2-First-
+1. Coordinators may compromise the availability of the system by refusing to validate the attestation and refusing to supply their key share.
+
+
+#### 1.6.3 Mitigations
+
+1. Helper party networks with a larger number of coordinators could utilize a threshold key which requires less than all key shares to decrypt the data, preventing an individual coordinator from compromising availability. This comes at the cost of more complicated analysis about the ability of an attacker to control different subsets of that helper party network.
+
+
+### 1.7. Helper party collusion
+
+If enough helper parties collude (beyond the proposal specific subset which an attacker is assumed to control), then none of the properties of the system hold. Such scenarios are outside the threat mode.
+
+However, we do assume that an attacker can always control at least one helper party (i.e., there are no perfectly trusted helper parties.)
+
+
+### 1.8 Cloud Providers for Helper Parties
+
+Helper parties may run either on physical machines owned by directly by the aggregator or (more commonly) subcontract with a cloud provider. We assume that an attacker can control some subset of cloud providers.
+
+
+#### 1.8.1 Assets
+
+1. All the assets of the helper party(ies) utilizing the cloud provider.
+2. All the assets of from the first and third parties.
+3. Identity of the helper parties.
+
+#### 1.8.2 Capabilities
+
+1. Cloud providers have all the capabilities of the helper parties utilizing that cloud provider.
+2. If a sufficient number of helper parties utilize a common cloud provider,
+    a. for aggregator networks, it can reconstruct the client/user assets;
+    b. for coordinator networks, it can reconstruct the decryption key.
+
+
+#### 1.8.3 Mitigations
+
+1. Helper parties networks should utilize sufficiently distinct cloud providers beyond the proposal specific subset which the attacker is assumed to control.
+
+
+### 1.9 Operators of TEEs
+
+As a piece of hardware, TEEs will have an operator with access to the machine. Most commonly, this will be a cloud provider. Depending on the specific hardware, there may be known vulnerabilities in which an attacker who only controls the operator can violate the obliviousness of client/user data. These attacks are outside this threat model, but are likely to inform specific web platform decisions about which instantiations of private computation to support.
+
+#### 1.9.1 Assets
+TODO
 
 
-#### 1.6.1. Capabilities
+#### 1.9.2 Capabilities
+TODO
 
 
+#### 1.9.2 Mitigations
+TODO
+
+
+### 1.10 TEE Manufacturers
+
+TEEs can provide "attestation" which verifies that the TEE is running in the expected state and running the expected code. These attestations are typically produced with an asymmetric key, where the private key is physically embedded into the TEE, and the public key is published via a system similar to a certificate authority.
+
+#### 1.9.1 Assets
+
+1. Private keys embedded in TEEs and their corresponding public keys.
```

Comment:
It is quite possible that operational practices at the TEE manufacturer is good enough that they don't have access to the private keys, but they will have access to the "root" key.

We should assume that they have both.

Reply:
Good callout on the "root" key, will add.

The access to the private key likely differs depending on what type of TEE is used, and I know that the intention in most cases is to not have access to it, but since we can't verify I included it here. Would it help to say something more like "While TEE Manufacturers take measures to prevent the leakage of private keys to themselves, we assume that they may have access to these private keys embedded in the TEEs."?

Reply:
Opened #26